iocore/cache/test/main.cc

```diff
@@ -0,0 +1,274 @@
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#define CATCH_CONFIG_MAIN
+#include "main.h"
+
+#define THREADS 1
+#define DIAGS_LOG_FILE "diags.log"
+
+char working_dir[1024] = {0};
+void
+test_done()
+{
+  shutdown_event_system = true;
+}
+
+const char *GLOBAL_DATA = (char *)ats_malloc(10 * 1024 * 1024 + 3); // 10M
+
+struct EventProcessorListener : Catch::TestEventListenerBase {
+  using TestEventListenerBase::TestEventListenerBase; // inherit constructor
+
+  virtual void
+  testRunStarting(Catch::TestRunInfo const &testRunInfo) override
+  {
+    getcwd(working_dir, 1024);
+    BaseLogFile *base_log_file = new BaseLogFile("stderr");
+    diags                      = new Diags(testRunInfo.name.c_str(), "*" /* tags */, "" /* actions */, base_log_file);
+    diags->activate_taglist("cache.*|agg.*|locks", DiagsTagType_Debug);
+    diags->config.enabled[DiagsTagType_Debug] = true;
+    diags->show_location                      = SHOW_LOCATION_DEBUG;
+
+    mime_init();
+    Layout::create();
+    RecProcessInit(RECM_STAND_ALONE);
+    LibRecordsConfigInit();
+    ink_net_init(ts::ModuleVersion(1, 0, ts::ModuleVersion::PRIVATE));
+    ink_assert(GLOBAL_DATA != nullptr);
+
+    netProcessor.init();
+    eventProcessor.start(THREADS);
+
+    ink_aio_init(AIO_MODULE_PUBLIC_VERSION);
+
+    EThread *thread = new EThread();
+    thread->set_specific();
+    init_buffer_allocators(0);
+
+    Layout::get()->sysconfdir = std::string_view("./test");
+    Layout::get()->prefix     = std::string_view("./test");
+    ::remove("./test/var/trafficserver/cache.db");
+  }
+};
+CATCH_REGISTER_LISTENER(EventProcessorListener);
+
+extern Store theCacheStore;
+
+void
+init_cache(size_t size, const char *name)
+{
+  ink_cache_init(ts::ModuleVersion(1, 0, ts::ModuleVersion::PRIVATE));
+  cacheProcessor.start();
+}
+
+void
+build_hdrs(HTTPInfo &info, const char *url, const char *content_type)
+{
+  HTTPHdr req;
+  HTTPHdr resp;
+  HTTPParser parser;
+  int err           = -1;
+  char buf[1024]    = {0};
+  const char *start = buf;
+  char *p           = buf;
+
+  REQUIRE(url != nullptr);
+
+  p += sprintf(p, "GET %s HTTP/1.1\n", url);
+  p += sprintf(p, "Host: www.scw00.com\n");
```

Comment:
It is better to define DEFAULT_URL in main.h, just like you do it in CacheTestHandler.h

```
#define DEFAULT_URL "http://www.scw00.com/"
```

Remove the Host header and make a explicit http request:
```
p += sprintf(p, "GET %s HTTP/1.1\n", DEFAULT_URL);
```

Reply:
will do.